### PR TITLE
TEST/ASYNC: Add retries to all tests

### LINKED ISSUE #4996: Failure in thread_mutex/test_async.modify_event/0
https://dev.azure.com/ucfconsort/0b36e3f0-8ab9-4a48-b68b-4b2350e02c88/_apis/build/builds/5303/logs/75
```
2020-04-08T10:04:02.4130883Z [ RUN      ] thread_mutex/test_async.two_timers/0 <thread_mutex>
2020-04-08T10:04:02.5738364Z [       OK ] thread_mutex/test_async.two_timers/0 (161 ms)
2020-04-08T10:04:02.5740641Z [ RUN      ] thread_mutex/test_async.modify_event/0 <thread_mutex>
2020-04-08T10:04:02.6141820Z /scrap/azure/agent-11/AZP_WORKSPACE/1/s/contrib/../test/gtest/ucs/test_async.cc:575: Failure
2020-04-08T10:04:02.6144264Z Expected: (le.count()) >= (1), actual: 0 vs 1
2020-04-08T10:04:04.6962746Z [  FAILED  ] thread_mutex/test_async.modify_event/0, where GetParam() = thread_mutex (2122 ms)
2020-04-08T10:04:04.6965819Z [----------] 4 tests from thread_mutex/test_async (2284 ms total)
2020-04-08T10:04:04.6967085Z 
2020-04-08T10:04:04.6968684Z [----------] 4 tests from rc_mlx5/uct_p2p_am_test
2020-04-08T10:04:04.6970218Z [ RUN      ] rc_mlx5/uct_p2p_am_test.am_async_response/0 <rc_mlx5/mlx5_0:1>
2020-04-08T10:04:04.7256026Z [     SKIP ] (!check_caps(UCT_IFACE_FLAG_CB_SYNC | UCT_IFACE_FLAG_CB_ASYNC, UCT_IFACE
```

### PR DESCRIPTION
**Pull request by yosefe**
# Why
Fix #4996

### CODE REVIEW

File: test/gtest/ucs/test_async.cc [hunk 1]
```diff
@@ -249,8 +249,7 @@ public ucs::test_base {
 protected:
     static const int      COUNT           = 40;
     static const unsigned SLEEP_USEC      = 1000;
-    static const int      EVENT_RETRIES   = 10;
-    static const int      TIMER_RETRIES   = 100;
+    static const int      NUM_RETRIES     = 100;
```
Comment (dmitrygx) [inline test/gtest/ucs/test_async.cc:252]: minor: maybe `unsigned` better?

**Comment by yosefe** [comment]
bot:pipe:retest

File: test/gtest/ucs/test_async.cc (hunk 1 above)
Comment (yosefe) [reply test/gtest/ucs/test_async.cc:252]: all counters in this test are 'int'...

**Pull request merged**